services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -817,6 +816,56 @@ class FlowCrudSpec extends HealthCheckSpecification {
         flows.each { flowHelper.deleteFlow(it) }
     }
 
+    def "System takes isl time_unstable info into account while creating a flow"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.paths.unique { it.first().portNo }.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Detect two available paths"
+        def firstPath = switchPair.paths.min { it.size() }
+        def secondPath = switchPair.paths.findAll { it != firstPath }.max { it.size() }
+        def altPaths = switchPair.paths.findAll {
+            it != firstPath && it.first().portNo != firstPath.first().portNo &&
+                    it != secondPath && it.first().portNo != secondPath.first().portNo
+        }.sort { it.size() }
+
+        and: "Make all alternative paths are unavailable (bring ports down on the srcSwitch)"
+        List<PathNode> broughtDownPorts = []
+        altPaths.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(antiflapMin + WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        and: "Make the first path unstable by bringing port down/up"
+        def islToBreak = pathHelper.getInvolvedIsls(firstPath).first()
+        northbound.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(WAIT_OFFSET) { assert islUtils.getIslInfo(islToBreak).get().state == FAILED }
+        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(WAIT_OFFSET) { assert islUtils.getIslInfo(islToBreak).get().state == DISCOVERED }
+
+        when: "Create a flow"
+        def flow = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created on the link from the second path"
+        PathHelper.convert(northbound.getFlowPath(flow.id)).first().portNo == secondPath.first().portNo
+
+        and: "Restore topology, delete the flow and reset costs"
+        broughtDownPorts.each { northbound.portUp(it.switchId, it.portNo) }
+        flowHelper.deleteFlow(flow.id)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != FAILED }
+        }
+        database.resetCosts()
```

Comment:
Is the ISL still in 'unstable' status at this point? Is there a way to get its current 'stability' status?

Reply:
Actually there is no way to get status of ISL via API